### DISCUSSION TITLE
Importing git histories for projects

### DISCUSSION THREAD

**Discussion by mmJordanO** [vercel/turborepo#284] [Help]
Hello.  I was very impressed on how easy it was to set up turborepo.

Our team wants to know if there's an easy way to go about adding an existing project into the monorepo and then import its git history along with it?  I wasn't sure how exactly this would behave however.  Would it be something like the monorepo has its own history and then the imported project has a separate history or are they somehow merged?

Thanks in advance.

**Comment by weyert** [comment]
Yeah, I am having the same question. I am not sure what the best approach is. I once asked in a relevant git tool a similar question but I have to admit I didn't fully understand the author's answer. You can see my question here: https://github.com/newren/git-filter-repo/discussions/192

**Reply by sephi-dev** [reply]
I did it using Lerna in the first place with preserve commit and flatten flag.

**Comment by kumavis** [comment]
I expected this to be a basic feature of a monorepo mgmt tool. topic seemingly not even covered in docs?
